login/service.go

```diff
@@ -756,6 +705,32 @@ func (keycloak *KeycloakOAuthProvider) CreateOrUpdateKeycloakUser(accessToken st
 	return identity, user, nil
 }
 
+func checkApproved(ctx context.Context, profileService UserProfileService, accessToken string, profileEndpoint string) (bool, error) {
+	profile, err := profileService.Get(accessToken, profileEndpoint)
+	if err != nil {
+		return false, err
+	}
+	if profile.Attributes == nil {
+		log.Warn(ctx, map[string]interface{}{
+			"username": profile.Username,
+		}, "no attributes found in the user's profile")
+		return false, nil
+	}
+	attributes := *profile.Attributes
+	approved := attributes[ApprovedAttributeName]
+	if approved == nil || len(approved) == 0 {
+		log.Warn(ctx, map[string]interface{}{
+			"username": profile.Username,
+		}, "no approved attribut found in the user's profile or the value is empty")
+		return false, nil
+	}
+	b, err := strconv.ParseBool(approved[0])
+	if err != nil {
+		return false, err
+	}
+	return b, nil
```

Comment:
Yeah, it's how KC returns attributes. As an array. Even if it's just a string. It will be an array with the only member.